code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+			return FALSE
+
+		if(tc_input < 10)
+			to_chat(user, SPAN_WARNING("This really isn't a useful amount for sabotage."))
+			return FALSE
+
+		to_chat(user, SPAN_WARNING("You begin to insert the crystals into [src]..."))
+
+		if(!do_after(user, 6 SECONDS, src))
+			return FALSE
+
+		switch(tc_input)
+			if(10 to 24)
+				sabotaged = SHUNT_SABOTAGE_MINOR
+			if(25 to 49)
+				sabotaged = SHUNT_SABOTAGE_MAJOR
+			if(50 to INFINITY)
+				sabotaged = SHUNT_SABOTAGE_CRITICAL
+
+		sabotaged_amt = tc_input
+		TC.use(tc_input)
+		to_chat(user, SPAN_DANGER("You successfully sabotage [src] by inserting the crystals!"))
+		return TRUE
+	. = ..()
+
+/obj/machinery/ftl_shunt/core/physical_attack_hand(var/mob/user)
+	if(sabotaged)
+		var/mob/living/carbon/human/h_user = null
+		if (!istype(user, /mob/living/carbon/human))
+			return
+		h_user = user //this is used specifically in the do_after because electrocute_act() is not a moblevel proc.
+		var/skill_delay = user.skill_delay_mult(SKILL_ENGINES, 0.3)
+		if(!user.skill_check(SKILL_ENGINES, SKILL_BASIC))
+			to_chat(user, SPAN_DANGER("You are nowhere near experienced enough to stick your hand into that thing."))
+			return FALSE
+		to_chat(user, SPAN_NOTICE("You reach your hand inside of [src] and slowly begin to remove re-align the accelerator vanes..."))
+		if(!do_after(user, (4 SECOND * skill_delay), src))
+			to_chat(user, SPAN_WARNING("You try to pull your hand away from the vanes, but you touch a conductor!"))
+			h_user.electrocute_act(rand(150,250), src, def_zone = user.get_active_held_item_slot())
+			return FALSE
+		var/obj/item/stack/telecrystal/TC = new
+		TC.amount = sabotaged_amt
+		TC.forceMove(get_turf(user))
+		user.put_in_hands(TC)
+		to_chat(user, SPAN_NOTICE("You realign the accelerator vanes by removing some kind of crystal, preventing what could have been a catastrophe."))
+		sabotaged = null
+		sabotaged_amt = 0
+		return TRUE
+	. = ..()
+
+//Custom procs.
+//Finds fuel ports.
+/obj/machinery/ftl_shunt/core/proc/find_ports()
+	var/datum/extension/local_network_member/network = get_extension(src, /datum/extension/local_network_member)
+	var/datum/local_network/lan = network.get_local_network()
+	if(lan)
+		var/list/ports = lan.get_devices(/obj/machinery/ftl_shunt/fuel_port)
+		for(var/obj/machinery/ftl_shunt/fuel_port/FP in ports)
+			if(!FP.master)
+				FP.master = src
+				fuel_ports += FP
+
+//Starts the teleport process, returns 1-6, with 6 being the all-clear.
+/obj/machinery/ftl_shunt/core/proc/start_shunt()
+	var/shunt_distance
+	var/vessel_mass = ftl_computer.linked.get_vessel_mass()
+	var/shunt_turf = locate(shunt_x, shunt_y, GLOB.using_map.overmap_z)
+
+	if(stat & (BROKEN))
+		return FTL_START_FAILURE_BROKEN
+	if(stat & (NOPOWER))
+		return FTL_START_FAILURE_POWER
+
+	if(!length(fuel_ports)) //no fuel ports
+		find_ports()
+		if(!length(fuel_ports))
+			return FTL_START_FAILURE_OTHER
+
+	if(world.time <= cooldown)
```

Comment:
Probably slightly better for performance to do this simple comparison before the find_ports() checks above.

Reply:
This is so that it finds the fuel ports, and if there's still no ports, then returns a failure.